src/memory/primitives/felt.test.ts

```diff
@@ -0,0 +1,38 @@
+import { test, expect, describe } from 'bun:test';
+import { ConversionError, Felt } from './felt';
+
+describe('felt', () => {
+  test('should initialise a felt correctly and perform good conversions', () => {
+    const felt = new Felt(10n);
+
+    expect(felt.getInner()).toEqual(10n);
```

Comment:
maybe we should implement a `.eq` then instead of this `getInner`

Reply:
we'd still need get getInner to perform the equality check 
```
eq(other: Felt): boolean {
    return this.inner == other._getInner();
  }
```